src/pytorch_lightning/plugins/precision/apex_amp.py

```diff
@@ -76,7 +76,7 @@ def backward(
 
     def optimizer_step(
         self,
-        model: Union["pl.LightningModule", Module],
+        model: Optional[Union["pl.LightningModule", Module]],
```

Comment:
Because it inherits from ```MixedPrecisionPlugin``` which inherits from ```PrecisionPlugin```. ```PrecisionPlugin``` requires model to be Optional otherwise the 2 first configurations of [this test](https://github.com/Lightning-AI/lightning/blob/03cbca430d7d82756b2786022cddcf767d3ffb89/tests/tests_pytorch/lite/test_parity.py#L118) does not run.

Reply:
I get the point of inheritance but not why the test would fail, the model there is always defined as a boringmodel?

Reply:
Ok got it. It's because of Lite where the model is optional and only passed to optimizer_step in case of deepspeed. So in general it remains optional.